### PR TITLE
change collection usages to seqeunce for performance

### PR DESCRIPTION
**Pull request by neetopia**

### CODE REVIEW

**Comment by neetopia** [comment]
Convert to draft as there is a output change in test `asMemberOf` that I couldn't understand at this moment.

File: api/src/main/kotlin/com/google/devtools/ksp/processing/SymbolProcessor.kt [hunk 1]
```diff
@@ -47,7 +47,7 @@ interface SymbolProcessor {
      * @param resolver provides [SymbolProcessor] with access to compiler details such as Symbols.
      * @return A list of deferred symbols that the processor can't process.
      */
-    fun process(resolver: Resolver): List<KSAnnotated>
```
Comment (neetopia) [inline api/src/main/kotlin/com/google/devtools/ksp/processing/SymbolProcessor.kt:50]: remain as list

File: compiler-plugin/src/main/kotlin/com/google/devtools/ksp/processing/impl/ResolverImpl.kt [hunk 2]
```diff
@@ -891,14 +891,14 @@ internal fun KSAnnotated.findAnnotationFromUseSiteTarget(): Collection<KSAnnotat
...
-                annotationsFromParents.addAll(it.originalAnnotations.filter { it.useSiteTarget == AnnotationUseSiteTarget.SETPARAM })
+                annotationsFromParents.addAll(it.originalAnnotations.asSequence().filter { it.useSiteTarget == AnnotationUseSiteTarget.SETPARAM })
                 parent = (parent as KSPropertyAccessorImpl).receiver
             }
             (parent as? KSPropertyDeclarationImpl)?.let {
-                annotationsFromParents.addAll(it.originalAnnotations.filter { it.useSiteTarget == AnnotationUseSiteTarget.SETPARAM })
+                annotationsFromParents.addAll(it.originalAnnotations.asSequence().filter { it.useSiteTarget == AnnotationUseSiteTarget.SETPARAM })
             }
-            annotationsFromParents
+            annotationsFromParents.asSequence()
```
Comment (ting-yuan) [inline compiler-plugin/src/main/kotlin/com/google/devtools/ksp/processing/impl/ResolverImpl.kt:900]: // TODO: eliminate `annotationsFromParents` and compute the Sequence on the fly.

File: compiler-plugin/src/main/kotlin/com/google/devtools/ksp/processor/ImplicitElementProcessor.kt [hunk 3]
```diff
@@ -60,11 +60,11 @@ class ImplicitElementProcessor : AbstractTestProcessor() {
         val comp2 = dataClass.declarations.single { it.simpleName.asString() == "comp2" } as KSPropertyDeclaration
         comp2.getter?.let { result.add("comp2.get(): ${it.origin}") }
         comp2.setter?.let { result.add("comp2.set(): ${it.origin}") }
-        val annotationType = comp1.getter?.let { result.add(it.annotations[0].annotationType.resolve().declaration.qualifiedName!!.asString()) }
+        val annotationType = comp1.getter?.let { result.add(it.annotations.toList()[0].annotationType.resolve().declaration.qualifiedName!!.asString()) }
```
Comment (ting-yuan) [inline compiler-plugin/src/main/kotlin/com/google/devtools/ksp/processor/ImplicitElementProcessor.kt:63]: nit: replace `toList()[0]` with `first()`

File: compiler-plugin/src/main/kotlin/com/google/devtools/ksp/symbol/impl/kotlin/KSTypeParameterImpl.kt [hunk 4]
```diff
@@ -52,9 +52,9 @@ class KSTypeParameterImpl private constructor(val ktTypeParameter: KtTypeParamet
         }
     }
 
-    override val bounds: List<KSTypeReference> by lazy {
-        val list = mutableListOf(ktTypeParameter.extendsBound)
-        list.addAll(
+    override val bounds: Sequence<KSTypeReference> by lazy {
+        val list = sequenceOf(ktTypeParameter.extendsBound)
+        list.plus(
```
Comment (ting-yuan) [inline compiler-plugin/src/main/kotlin/com/google/devtools/ksp/symbol/impl/kotlin/KSTypeParameterImpl.kt:58]: `plus` has no side effect.

**Review by ting-yuan — COMMENTED** [review]
LGTM except one use of `plus`.

**Review by ting-yuan — APPROVED** [review]

**Pull request merged**